src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -138,8 +149,8 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
           throw cudf::jni::jni_exception("injected RetryOOM");
         }
 
-        if (thread->second.split_and_retry_oom_injected) {
-          thread->second.split_and_retry_oom_injected = false;
+        if (thread->second.split_and_retry_oom_injected > 0) {
+          thread->second.split_and_retry_oom_injected--;
           JNIEnv *env = cudf::jni::get_jni_env(jvm);
           // TODO cache what is needed for this...
```

Comment:
Let me push a commit to handle this TODO

Reply:
https://github.com/NVIDIA/spark-rapids-jni/pull/950/commits/9b4b1219bd017f549ea97a2abe940e320881cd7c